lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -885,22 +885,30 @@ LinearLayout wmmaDotOperandToLinearLayout(DotOperandEncodingAttr dotWmmaLayout,
   auto laneOrder =
       getOrderForDotOperand(dotWmmaLayout.getOpIdx(), rank, /*kContig*/ true);
   // generate continuous part of register bases(i.e. kWidth)
-  std::vector<std::vector<int32_t>> registerBase;
   const int32_t kWidth = dotWmmaLayout.getKWidth();
-  for (int i = 1; i < kWidth; i *= 2)
-    registerBase.push_back(std::vector<int32_t>{i, 0});
-  std::vector<std::vector<int32_t>> laneBase = {{0, 1}, {0, 2}, {0, 4}, {0, 8}};
-  switch (wmmaLayout.getVersion()) {
-  case 1:
+  auto version = wmmaLayout.getVersion();
+  std::vector<std::vector<int32_t>> registerBase, laneBase;
+  if (version == 1 || version == 2) {
+    for (int i = 1; i < kWidth; i *= 2)
+      registerBase.push_back(std::vector<int32_t>{i, 0});
     // WMMA version 1 duplicates values in lanes 0-15 and 16-31
-    laneBase.push_back({0, 0});
-    break;
-  case 2:
     // WMMA version 2 offset values in lanes 0-15 and 16-31 across k dimensions
-    laneBase.push_back({kWidth, 0});
-    break;
-  default:
-    assert(false && "unexpected version");
+    laneBase = {{0, 1}, {0, 2}, {0, 4}, {0, 8}, {version == 1 ? 0 : kWidth, 0}};
+  } else {
+    assert(version == 3 && "unexpected wmma version");
+    if (kWidth == 16) {
+      registerBase =
+          std::vector<std::vector<int32_t>>{{1, 0}, {2, 0}, {4, 0}, {16, 0}};
```

Comment:
This is odd. How is it that `kWidth=16` has 32 contiguous elements?

Reply:
Yeah. 0-7 and 16-23 to be precise. 8 contiguous elements.

Reply:
Ok I changed `kWidth` to 8 and added `instrShape`. 

@antiagainst Please take another look. 

Also note that for almost all gfx11 and gfx12 wmma, we are using 16x16x16 wmma, expect for i4 on gfx12 which uses 16x16x32.